### DISCUSSION TITLE
Issue using external tomo file for plane wave simulation

### DISCUSSION THREAD

**Discussion by kenenig** [SPECFEM/specfem3d#1694] [SPECFEM2D General]
I was having issue running the tomographic_ocean_model example. It give back an immediate backtrace error when I run it either serially or with multiple processors. I am using an Macbook with a M1 processor so my guess is that there is something wrong when first compiling Specfem but I am not sure. When I tried creating my own tomographic model and running a similar plane wave simulation, the tomo file was read successfully but returned the same backtrace error when running the internal mesher. Has anyone else had a similar issue using an external tomography file?

<img width="1274" alt="Screen Shot 2024-04-25 at 12 20 15 PM" src="https://github.com/SPECFEM/specfem3d/assets/140196629/f3222070-587a-4904-9be7-241755317acb">
<img width="1274" alt="Screen Shot 2024-04-25 at 12 36 29 PM" src="https://github.com/SPECFEM/specfem3d/assets/140196629/e54d891a-3b31-421f-a1c4-28ca9aa55148">

**Comment by homnath** [comment]
Which version are you using? It works fine with the recent devel branch. But I tried on Linux and WINDOWS.

**Reply by kenenig** [reply]
I am running 8.1.0 version and I am using the master branch. I just looked into the devel branch too but it said that it had issues on Mac. 

As an update, I am able to run this example. However, for the simulation which has my external tomography file but a different issue occurs. The tomography file appears to be successfully read (see screenshot below) but the last velocity is the only one used and the material is made homogenous. I was trying to test if I can use a tomography file for a non-plane wave simulation and it was giving me the same issue. I included my Par_file, SOURCE, and external tomography file used for the run in the attached zip file for context.   
<img width="913" alt="SF_tomo_terminal_output" src="https://github.com/SPECFEM/specfem3d/assets/140196629/9c22140c-3ff5-4b90-8fb4-f7ac8e8bd121">

[SF_tomo_issue.zip](https://github.com/SPECFEM/specfem3d/files/15257323/SF_tomo_issue.zip)

**Comment by homnath** [comment]
How did you find, it is using the homogeneous model? Can you also upload all mesh files?


---------------------------------------
Hom Nath Gharti, PhD
Assistant Professor | Digital Earth Scientist
https://www.digitalearthscience.com/
Department of Geological Sciences and Geological Engineering
Miller Hall 314, 36 Union St
Queen’s University
Kingston, ON K7L 3N6, Canada


Queen’s University is situated on traditional
Anishinaabe and Haudenosaunee Territory.<https://www.queensu.ca/encyclopedia/t/traditional-territories>
________________________________

**Reply by kenenig** [reply]
Yes here they are. I used the plot_model_from_gll_or_binary_file.py script to plot the vs, vp, and rho and they were all one number. 

[mesh_0000.zip](https://github.com/SPECFEM/specfem3d/files/15264113/mesh_0000.zip)
[mesh_0001.zip](https://github.com/SPECFEM/specfem3d/files/15264129/mesh_0001.zip)
[mesh_0002.zip](https://github.com/SPECFEM/specfem3d/files/15264134/mesh_0002.zip)
[mesh_0003_rho_vp.zip](https://github.com/SPECFEM/specfem3d/files/15264215/mesh_0003_rho_vp.zip)
[mesh_0003vs_0004rho.zip](https://github.com/SPECFEM/specfem3d/files/15264217/mesh_0003vs_0004rho.zip)
[mesh_0004_vs_vp.zip](https://github.com/SPECFEM/specfem3d/files/15264206/mesh_0004_vs_vp.zip)

**Reply by homnath** [reply]
I see. Can you send MOR1_interfaces_outsideonly.dat file?

**Reply by kenenig** [reply]
Yes, here it is. I only defined the outer portions of the model space while troubleshooting. [MOR1_interfaces_outsideonly.dat.zip](https://github.com/SPECFEM/specfem3d/files/15265320/MOR1_interfaces_outsideonly.dat.zip)

**Reply by homnath** [reply]
![forward_image000000005](https://github.com/SPECFEM/specfem3d/assets/5777522/b3305beb-6314-4a6c-8ce7-f28eeb5aba3d)

I think it is correctly implemeting the tomographic model. Does your model resemble the one shown in the snapshot?

**Reply by homnath** [reply]
![forward_image000000700](https://github.com/SPECFEM/specfem3d/assets/5777522/8a72d09a-0c29-420c-b993-8265b80cdb06)
